CoffeeMachineSimulator/CoffeeMachineSimulator.Services/Services/CoffeeService.cs

```diff
@@ -19,13 +19,18 @@ public CoffeeService()
         public void AddCoffee(CoffeeModel coffeeToAdd)
         {
             if (coffeeToAdd == null) throw new Exception("You should not add null entries!");
-
-            coffeeModels.Add(coffeeToAdd);
+            
+            if (coffeeModels.Any(i => i == coffeeToAdd)) throw new Exception("Already exists an entry with the same Id");
+            if (!(coffeeToAdd.Id == Guid.Empty || coffeeToAdd.Name == null || coffeeToAdd.Price <= 0.0f || coffeeModels.Any(i => i.Name == coffeeToAdd.Name) || coffeeModels.Any(j => j.Id == coffeeToAdd.Id) || coffeeModels.Any(k => k.Price == coffeeToAdd.Price)))
```

Comment:
please make a separate method which returns true if the coffee to add already exists in the list of coffees. Also, please be consistent when using lambda expressions (you are using i, j, k which can be confusing)

Reply:
Please check if that fulfills your requirements (Ex:  coffeeModels.Any(k => k.Price == coffeeToAdd.Price) can return true, there can be 2 coffees with the same price)